railties/lib/rails/command.rb

```diff
@@ -0,0 +1,96 @@
+require "active_support"
+require "active_support/core_ext/enumerable"
+require "active_support/core_ext/object/blank"
+require "active_support/core_ext/hash/transform_values"
+
+require "thor"
+
+module Rails
+  module Command
+    autoload :Behavior, "rails/command/behavior"
+    autoload :Base, "rails/command/base"
+
+    include Behavior
+
+    class << self
+      def hidden_commands # :nodoc:
+        @hidden_commands ||= []
+      end
+
+      def environment
```

Comment:
Yeah, executing a command in a given environment is probably a Rails internal thing. On the other hand having every command except a `--environment` could be neat (which this method could help assign the default for).